docs/en/index.md

```diff
@@ -159,33 +159,190 @@ Nested GridFields
+
+        return $form;
+    }
+}
+```
+
+There are several ways to use Nested GridField. The implementation depends on the type of data you will be using in your GridField.
+For instance, if there is a DataObject that has the [`Hierarchy`](api:SilverStripe\ORM\Hierarchy\Hierarchy) extension, you can use the following approach.
+
+As an example, here we can use a typical hierarchy of Security Groups, where another Group can serve as a parent.
+
+```php
+class MySecurityAdminExtension extends Extension
+{
+    public function updateGridFieldConfig($config)
+    {
+        if ($this->owner->getModelClass() === Group::class) {
+            $config->addComponent(GridFieldNestedForm::create());
+        }
+    }
+}
+```
+
+Or also view a list of all members of a given group.
+
+```php
+class MySecurityAdminExtension extends Extension
+{
+    public function updateGridFieldConfig($config)
+    {
+        if ($this->owner->getModelClass() === Group::class) {
+            $config->addComponent(GridFieldNestedForm::create()->setRelationName('Members'));
+        }
+    }
+}
+```
+
+```yml
+SilverStripe\Admin\SecurityAdmin:
+  extensions:
+    - MyApp\MyExtensions\MySecurityAdminExtension
+```
+
+Another way to use Nested GridField together with DataObjects that do not have the `Hierarchy` extension but have `has_many` relationships with other objects.
+Let's say you have the following DataObject that has multiple levels of relationships, and an Admin section where the data of this object will be displayed.
+You want the user to be able to view information regarding this object and all nested objects on the same page, without the need to navigate to each object individually.
+In this case, you can use the following approach.
+
 ```php
-// Basic usage, defaults to the Children-method for Hierarchy objects.
-$grid->getConfig()->addComponent(GridFieldNestedForm::create());
+class ParentNode extends DataObject
+{
+    ...
+
+    private static $has_many = [
+        'ChildNodes' => BranchNode::class,
+    ];
+}
 
-// Usage with custom relation
-$grid->getConfig()->addComponent(GridFieldNestedForm::create()->setRelationName('MyRelation'));
 ```
 
-You can define your own custom GridField config for the nested GridField configuration by implementing a `getNestedConfig`
-on your nested model (should return a `GridField_Config` object).
 ```php
-class NestedObject extends DataObject
+class ChildNode extends DataObject
 {
-	private static $has_one = [
-		'Parent' => ParentObject::class
+    ...
+
+    private static $has_one = [
+        'ParentNode' => ParentNode::class,
+    ];
+
+    private static $has_many = [
+        'GrandChildNodes' => GrandChildNode::class,
+    ];
+
+    public function getNestedConfig(): GridFieldConfig
+    {
+        $config = new GridFieldConfig_RecordEditor();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('GrandChildNodes'));
+        return $config;
+    }
+}
+
+```
+
```

Comment:
Please add a paragraph here about the `getNestedConfig()` method.

Reply:
Updated